js_modules/dagster-ui/packages/ui-core/src/assets/CalculateChangedAndMissingDialog.tsx

```diff
@@ -0,0 +1,205 @@
+import {useQuery, gql} from '@apollo/client';
+import {
+  Spinner,
+  Dialog,
+  DialogBody,
+  DialogFooter,
+  Button,
+  Box,
+  Icon,
+  Colors,
+  Checkbox,
+  MiddleTruncate,
+} from '@dagster-io/ui-components';
+import {useVirtualizer} from '@tanstack/react-virtual';
+import React from 'react';
+import {Link} from 'react-router-dom';
+import styled from 'styled-components';
+
+import {showCustomAlert} from '../app/CustomAlertProvider';
+import {displayNameForAssetKey} from '../asset-graph/Utils';
+import {Container, Inner, Row} from '../ui/VirtualizedTable';
+
+import {useMaterializationAction} from './LaunchAssetExecutionButton';
+import {isAssetStale, isAssetMissing} from './Stale';
+import {asAssetKeyInput} from './asInput';
+import {assetDetailsPathForKey} from './assetDetailsPathForKey';
+import {AssetKey} from './types';
+import {
+  AssetStaleStatusQuery,
+  AssetStaleStatusQueryVariables,
+} from './types/CalculateChangedAndMissingDialog.types';
+
+export const CalculateChangedAndMissingDialog = React.memo(
+  ({
+    isOpen,
+    onClose,
+    assets,
+    preferredJobName,
+  }: {
+    isOpen: boolean;
+    assets: {
+      assetKey: AssetKey;
+    }[];
+    onClose: () => void;
+    preferredJobName?: string;
+  }) => {
+    const {data, loading, error} = useQuery<AssetStaleStatusQuery, AssetStaleStatusQueryVariables>(
+      ASSET_STALE_STATUS_QUERY,
+      {
+        variables: {
+          assetKeys: assets.map(asAssetKeyInput),
+        },
+        skip: !isOpen,
+      },
+    );
+    const {
+      onClick,
+      loading: materializeActionLoading,
+      launchpadElement,
+    } = useMaterializationAction(preferredJobName);
```

Comment:
Could we have this dialog pass the selected keys back (Eg: `onClose(selectedAssetKeys)`) and then in the LaunchAssetExecutionButton call `onClick(selectedAssetKeys, e)` to thread this back into the existing flow?

It's a bit odd that the `CalculateChangedAndMissingDialog` uses a second copy of this hook to launch the assets all by itself when it's parent already had this hook wired up, would be nice to not render a second launchpadElement / not pass `preferredJob` into this dialog. Not a blocker though

Reply:
Ah I was thinking the dialog might be usable outside of the button by some other context but if we won't do that then I could tie it back to the button